test/e2e/networkpolicy_test.go

```diff
@@ -806,6 +806,107 @@ func TestIngressPolicyWithoutPortNumber(t *testing.T) {
 	}
 }
 
+func TestIngressPolicyWithEndPort(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	serverPort := int32(80)
+	serverEndPort := int32(84)
+	policyPort := int32(81)
+	policyEndPort := int32(83)
+
+	var serverPorts []int32
+	for i := serverPort; i <= serverEndPort; i++ {
+		serverPorts = append(serverPorts, i)
+	}
+
+	serverName, serverIPs, cleanupFunc := createAndWaitForPodWithMultiPorts(t, data, data.createAgnhostPodOnNodeWithMultiPort, "test-server-", "", serverPorts)
+	defer cleanupFunc()
+
+	clientName, _, cleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", "")
+	defer cleanupFunc()
+
+	preCheckFunc := func(serverIP string) {
+		// The client can connect to server on all ports.
+		for _, port := range serverPorts {
+			if err = data.runNetcatCommandFromTestPod(clientName, serverIP, port); err != nil {
+				t.Fatalf("Pod %s should be able to connect %s, but was not able to connect", clientName, net.JoinHostPort(serverIP, fmt.Sprint(port)))
+			}
+		}
+
+	}
+
+	if clusterInfo.podV4NetworkCIDR != "" {
+		preCheckFunc(serverIPs.ipv4.String())
+	}
+	if clusterInfo.podV6NetworkCIDR != "" {
+		preCheckFunc(serverIPs.ipv6.String())
+	}
+
+	protocol := corev1.ProtocolTCP
+	spec := &networkingv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"antrea-e2e": serverName,
+			},
+		},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		Ingress: []networkingv1.NetworkPolicyIngressRule{
+			{
+				Ports: []networkingv1.NetworkPolicyPort{
+					{
+						Protocol: &protocol,
+						Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: policyPort},
+						EndPort:  &policyEndPort,
+					},
+				},
+				From: []networkingv1.NetworkPolicyPeer{{
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"antrea-e2e": clientName,
+						},
+					}},
+				},
+			},
+		},
+	}
+	np, err := data.createNetworkPolicy("test-networkpolicy-ingress-with-endport", spec)
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(np); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	npCheck := func(serverIP string) {
+		for _, port := range serverPorts {
+			serverAddress := net.JoinHostPort(serverIP, fmt.Sprint(port))
+			err = data.runNetcatCommandFromTestPod(clientName, serverIP, port)
+			if port >= policyPort && port <= policyEndPort {
+				if err != nil {
+					t.Fatalf("Pod %s should be able to connect %s, but was not able to connect", clientName, serverAddress)
+				}
+			} else if err == nil {
+				t.Fatalf("Pod %s should be not able to connect %s, but was able to connect", clientName, serverAddress)
```

Comment:
these should be `t.Errorf` IMO: even in case of error, we can keep checking the other ports.

also maybe include the port number to the error message so there is more context in case of failure

Reply:
Done.